### DISCUSSION TITLE
Super Performance on Class Freeing

### DISCUSSION THREAD

**Discussion by Kr00l** [twinbasic/twinbasic#212] [Show and tell]
In VB6 it is quite known that Instantiation a bulk of class instances goes quite fast.
However, the class freeing begings to slow down progressively! So fast at the beginning but slows down dramatically..

Giving the below code as an example, the VB6 freeing times are as following: 
100k = 0.3902
200k = 2.2432
300k = 6.2895
and so on..

tB is currently super duper fast on this, which is exciting.
VB6 is so slow on this due to the IConnectionPoint-Containers and it's COM-Eventmechanism.

tB does here a direct callback to Class_Initialize and Class_Terminate ?

This is not a feature request or bug report. I just wanted to point this out as this is actually a good benefit !

Btw, the freeing time (in IDE) in tB for 300k is 0.0078

```
Class EnMass
' Void
End Class

Module ModHelloWorld

    ' This project type is set to 'Standard EXE' in the Settings file, so you need a Main() subroutine to run when the EXE is started.
    Public Sub Main()
        Const COUNT As Long = 300000 ' 300k
        Dim Arr() As EnMass, i As Long
        ReDim Arr(0 To COUNT) As EnMass
        For i = 0 To COUNT
            Set Arr(i) = New EnMass
        Next i

        Dim Time As Double
        Time = VBA.Timer

        Erase Arr()

        MsgBox VBA.Timer - Time
    End Sub
	
End Module
```

**Comment by WaynePhillipsEA** [comment]
Wow, VB6/VBA is really slow here.  I just had to double check that the tB version was working correctly (it is) as the difference is huge!  Thanks!

**Comment by KarlSevenSeven** [comment]
This is really VERY good!
I have some code parts which really suffer from the VB6 'speed'.
BTW, VB5 was much faster in this regard.

Thanks @Kr00l for testing it.
And @WaynePhillipsEA for doing so well.
